app/src/main/java/com/github/se/polyfit/ui/screen/OverviewScreen.kt

```diff
@@ -293,83 +157,73 @@ fun OverviewScreen(
-                                        hiltViewModel<GraphViewModel>().DateList(),
-                                        DisplayScreen.OVERVIEW))
-                          } else {
-                            Spacer(modifier = Modifier.fillMaxSize().testTag("LineChartSpacer"))
-                          }
-                        }
-                  }
-                }
+            ) {
+              Column(modifier = Modifier.fillMaxSize().testTag("Graph Card Column")) {
+                Text(
+                    text = "Calories Graph",
+                    style = MaterialTheme.typography.headlineLarge,
+                    fontWeight = FontWeight.Bold,
+                    color = MaterialTheme.colorScheme.secondary,
+                    modifier =
+                        Modifier.padding(start = 10.dp, top = 10.dp)
+                            .weight(1f)
+                            .testTag("Graph Card Title")
+                            .clickable { navigation.navigateToGraph() })
+                Box(
+                    modifier =
+                        Modifier.fillMaxSize(0.85f)
+                            .align(Alignment.CenterHorizontally)
+                            .testTag("Graph Box")
+                            .clickable { navigation.navigateToGraph() }) {
+                      if (!isTestEnvironment) {
+                        LineChart(
+                            modifier = Modifier.testTag("Overview Line Chart").fillMaxSize(),
+                            lineChartData =
+                                lineChartData(
+                                    hiltViewModel<GraphViewModel>().DataPoints(),
+                                    hiltViewModel<GraphViewModel>().DateList(),
+                                    DisplayScreen.OVERVIEW))
+                      } else {
+                        Spacer(modifier = Modifier.fillMaxSize().testTag("LineChartSpacer"))
+                      }
+                    }
+              }
+            }
           }
         }
   }
 }
+
+private fun callCamera(
+    context: Context,
+    startCamera: ManagedActivityResultLauncher<Intent, ActivityResult>,
+    requestPermissionLauncher: ManagedActivityResultLauncher<String, Boolean>
+): () -> Unit = {
+  // Check if the permission has already been granted
+  when (ContextCompat.checkSelfPermission(context, Manifest.permission.CAMERA)) {
+    PackageManager.PERMISSION_GRANTED -> {
+      // You can use the camera
+      val takePictureIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
+      try {
+        startCamera.launch(takePictureIntent)
+      } catch (e: Exception) {
+        // Handle the exception if the camera intent cannot be launched
+        Log.e("HomeScreen", "Error launching camera intent: $e")
+      }
+    }
+    else -> {
+      // Request the permission
+      requestPermissionLauncher.launch(Manifest.permission.CAMERA)
+    }
+  }
+}
```

Comment:
in my opinion this function could be moved to the viewModel

Reply:
Great idea. It makes way more sense to have that function in the viewmodel.